@here/harp-mapview/lib/BoundsGenerator.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+    EarthConstants,
+    GeoCoordinates,
+    mercatorTilingScheme,
+    ProjectionType,
+    TileKey
+} from "@here/harp-geoutils";
+import { Frustum, Line3, Matrix4, Plane, Vector2, Vector3 } from "three";
+
+import { TileGeometryCreator } from "./geometry/TileGeometryCreator";
+import { MapView } from "./MapView";
+import { MapViewUtils } from "./Utils";
+
+export class BoundsGenerator {
+    //TODO: check what we really need here
+    constructor(private m_mapView: MapView) {}
```

Comment:
Please don't pass whole mapview but only the things that are needed (i.e. projection and camera)

Reply:
I actually tried to to that, as it only needs the camera and the projection, but it has a dependency on MapViewUtils.rayCastGeoCoordinates, wich needs the mapView (although its also only using the camera and projection)

Reply:
I added the option to only pass camera and projection to the MapViewUtils.rayCastWorldCoordinates, therefore mapview is not needed anymore